src/status_im2/contexts/chat/bottom_sheet_composer/view.cljs

```diff
@@ -0,0 +1,137 @@
+(ns status-im2.contexts.chat.bottom-sheet-composer.view
+  (:require
+    [quo2.foundations.colors :as colors]
+    [react-native.core :as rn]
+    [react-native.gesture :as gesture]
+    [react-native.hooks :as hooks]
+    [react-native.reanimated :as reanimated]
+    [reagent.core :as reagent]
+    [utils.i18n :as i18n]
+    [status-im2.contexts.chat.bottom-sheet-composer.style :as style]
+    [status-im2.contexts.chat.bottom-sheet-composer.images.view :as images]
+    [utils.re-frame :as rf]
+    [status-im2.contexts.chat.bottom-sheet-composer.utils :as utils]
+    [status-im2.contexts.chat.bottom-sheet-composer.constants :as c]
+    [status-im2.contexts.chat.bottom-sheet-composer.actions.view :as actions]
+    [status-im2.contexts.chat.bottom-sheet-composer.keyboard :as kb]
+    [status-im2.contexts.chat.bottom-sheet-composer.sub-view :as sub-view]
+    [status-im2.contexts.chat.bottom-sheet-composer.effects :as effects]
+    [status-im2.contexts.chat.bottom-sheet-composer.gesture :as drag-gesture]
+    [status-im2.contexts.chat.bottom-sheet-composer.handlers :as handle]))
+
+(defn sheet
+  [insets window-height blur-height opacity background-y]
+  [:f>
+   (fn []
+     (let [props {:input-ref                   (atom nil)
+                  :keyboard-show-listener      (atom nil)
+                  :keyboard-frame-listener     (atom nil)
+                  :keyboard-hide-listener      (atom nil)
+                  :emoji-kb-extra-height       (atom nil)
+                  :saved-emoji-kb-extra-height (atom nil)}
+           state {:text-value            (reagent/atom "")
```

Comment:
`props` is not state, since these are normal atoms, so they won't affect the component's lifecycle. Maybe `props` is not the best name, feel free to share any alternatives names

Reply:
In this particular case `props` is a map of states. Atoms are states in Clojure, but okay, not in the sense of React states. I guess we both looked differently at the terminology ;)

To be more technically correct, the `props` binding is a map of non-reactive states and `state` is a map of reactive states. Both are states in Clojure, but one binding is made of reactive stuff and the other non-reactive.

It's fine as is @OmarBasem, my suggestion would be to not group states in maps, but I can see why you prefer the code as is.